climada/hazard/trop_cyclone.py

```diff
@@ -334,14 +347,17 @@ def _tc_from_track(self, track, centroids, coastal_idx, model='H08',
                                                              metric=metric)
         reachable_coastal_centr_idx = coastal_idx[reachable_centr_idx]
         npositions = windfields.shape[0]
-        intensity = np.zeros(ncentroids)
-        intensity[reachable_coastal_centr_idx] = np.linalg.norm(windfields, axis=-1)\
-                                                          .max(axis=0)
+
+        intensity = np.linalg.norm(windfields, axis=-1).max(axis=0)
         intensity[intensity < self.intensity_thres] = 0
+        intensity_sparse = sparse.csr_matrix(
+            (intensity, reachable_coastal_centr_idx, [0, intensity.size]),
+            shape=(1, ncentroids))
+        intensity_sparse.eliminate_zeros()
```

Comment:
As mentioned above, this introduces another local variable, which is against the spirit of the pylint-directive that restricts the number of local variables in a function.

Reply:
Indeed. But here I would argue that it is really needed. Variables changing types is worse than pylint complaining about too many variables imho.